server/userAuth.js

```diff
@@ -4,20 +4,47 @@ const passport = require('passport')
 const session = require('express-session')
 const md5 = require('md5')
 const GoogleStrategy = require('passport-google-oauth20')
+const SlackStrategy = require('passport-slack-oauth2').Strategy
 
 const log = require('./logger')
 const {stringTemplate: template} = require('./utils')
 
 const router = require('express-promise-router')()
 const domains = new Set(process.env.APPROVED_DOMAINS.split(/,\s?/g))
 
-passport.use(new GoogleStrategy.Strategy({
-  clientID: process.env.GOOGLE_CLIENT_ID,
-  clientSecret: process.env.GOOGLE_CLIENT_SECRET,
-  callbackURL: '/auth/redirect',
-  userProfileURL: 'https://www.googleapis.com/oauth2/v3/userinfo',
-  passReqToCallback: true
-}, (request, accessToken, refreshToken, profile, done) => done(null, profile)))
+const authStrategies = ['google', 'Slack']
+let authStrategy = process.env.OAUTH_STRATEGY
```

Comment:
I would advocate for throwing a `.toLowerCase()` on this to avoid potential confusion with capitalization, as @isaacwhite mentioned in his review.

Reply:
updated the ReadMe with the problems there - Passport wasn't recognizing a call to `Passport.authenticate` unless slack is capitalized